src/unix/process.c

```diff
@@ -424,20 +431,135 @@ int uv_spawn(uv_loop_t* loop,
 
   /* Acquire write lock to prevent opening new fds in worker threads */
   uv_rwlock_wrlock(&loop->cloexec_lock);
-  pid = fork();
 
-  if (pid == -1) {
-    err = UV__ERR(errno);
-    uv_rwlock_wrunlock(&loop->cloexec_lock);
-    uv__close(signal_pipe[0]);
-    uv__close(signal_pipe[1]);
-    goto error;
-  }
+#if defined(__APPLE__)
+  if (__builtin_available(macOS 10.15, *)) {
+    posix_spawnattr_t attrs;
+    err = posix_spawnattr_init(&attrs);
+    if(err)
+      goto error;
 
-  if (pid == 0) {
-    uv__process_child_init(options, stdio_count, pipes, signal_pipe[1]);
-    abort();
+    if ((options->flags & UV_PROCESS_SETUID) && posix_spawnattr_set_uid_np(&attrs, options->uid)) {
+      goto error;
+    }
+    if ((options->flags & UV_PROCESS_SETGID) && posix_spawnattr_set_gid_np(&attrs, options->gid)) {
+      goto error;
+    }
+
+    // Set flags for spawn behavior 
+    // 1) POSIX_SPAWN_CLOWEXEC_DEFAULT: (Apple Extension) All descriptors in t
+    // he parent will be treated as if they had been created with O_CLOEXEC. 
+    // The only fds that will be passed on to the child are those manipulated by the file actions
+    // 2) POSIX_SPAWN_SETSIGDEF: signals mentioned in sawn-sigdeffault in the spawn attributes
+    // will be reset to behave as their default
+    // 3) POSIX_SPAWN_SETSIGMASK: signal mask will be set to the value of spaw-sigmask in attributes
+    // 4) POSIX_SPAWN_SETSID: if process should be a new session leader, do that.
```

Comment:
```suggestion
    // Set flags for spawn behavior 
    // 1) POSIX_SPAWN_CLOWEXEC_DEFAULT: (Apple Extension) All descriptors in
    // the parent will be treated as if they had been created with O_CLOEXEC. 
    // The only fds that will be passed on to the child are those manipulated by the file actions
    // 2) POSIX_SPAWN_SETSIGDEF: Signals mentioned in posix_spawnattr_setsigdefault
    // will be reset to behave as their default
    // 3) POSIX_SPAWN_SETSIGMASK: Signal mask will be set to the value of posix_spawnattr_setsigmask
    // 4) POSIX_SPAWN_SETSID: Make the process a new session leader if a detached session was requested
```

Reply:
spawn-sigdefault and spawn-sigmask is how the manpage refers to those concepts so I'm keeping those, but changing all the rest.

Reply:
Gotcha, but at least fix the typos in them 😝